### DISCUSSION TITLE
Questions Regarding Implementation Details

### DISCUSSION THREAD

**Discussion by framewolf** [ANLAB-KAIST/KENSv3#179] [Project Q&A 2. Transport Layer (TCP)]
hello.
i have a few questions regarding the implementation details to ensure that I am following the guidelines correctly. 
1. Am I allowed to include the <algorithm> header?
2. Is it permissible to add member functions or new fields to the given classes in the TCPAssignment.hpp file?(such as class TCPAssignment, class TCPAssignmentProvider)
3. Are we allowed to use global variables in our implementation?

I want to make sure that my approach aligns with the assignment requirements and does not violate any rules. 
Thank you

**Comment by Pusnow** [comment]
1. Yes, you can include POSIX, Linux, C++17 headers.
2. Yes, but we only recommend add members in TCPAssignment.
3. Yes, but not recommended due to unwanted data sharing among hosts.
